### ISSUE TITLE
Failure when resolving classes across packages

### ISSUE THREAD

**Issue by twoseat** [immutables/immutables#390]
I have two packages:

`Alpha`, which contains `AbstractCharlie` and `Delta`. `AbstractCharlie` uses `Delta`

`Bravo`, which contains `Delta` and `Echo`. `Echo` extends `AbstractCharlie`

(So the two `Deltas` are different packages with the same name)

This fails with a conflict, because Immutables doesn't recognise that the reference to `Delta` in `AbstractCharlie` is from the `Alpha` package, not the `Bravo` package.

The workaround is to make the reference explicit, something like:
`public abstract org.example.Alpha.Delta getDelta()`
This works, but is undesirable.

(My apologies for the text-only explanation - I need an issue reference to include in my code so I can complete it, then I'll come back with code!)

**Comment by elucash** [comment]
I kinda know where this bug might come from, thank you for reporting this!

**Comment by elucash** [comment]
Just added a fix for this (soon to be released), and have couple of notes about it. While I believe this particular case might be solved by the fix, there's no reliable solution for this when types are not-yet-generated, and the current code contain a lot of guess-work to figure out the "most likely" way to resolve imports.
The solution will not work if generics are used for not-yet-generated types, or they will be used as generic arguments for some other types, or when star-imports are present, there's not much we can guess then.
Here are couple of workarounds if such or similar problems persist:

1) Having a source code of abstract value type referencing not-yet-generated type, and especially if it will be extended by other types, add assertive import to designate package of yet-to-be generated file

```java
package alpha;
import  alpha.Delta; //  <-- asserting that to be generated Delta coming from this package,
// this will be also used if inheriting AbstractCharlie's attributes
interface AbstractCharlie {
   Delta getDelta();
}
```

2) Avoid using inverse generation strategy in favor of the one shown at the top of the readme file (https://github.com/immutables/immutables/blob/master/README.md). When using such strategy, you use class without `Immutable` prefix (if that is the main advantage of inverse generation).

**Comment by elucash** [comment]
Released in 2.2.11 Thank you!
